projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -124,6 +131,19 @@ public boolean buildImage(
       String imageName, DataJob dataJob, JobDeployment jobDeployment, Boolean sendNotification)
       throws ApiException, IOException, InterruptedException {
 
+    String awsSecretAccessKey = this.awsSecretAccessKey;
```

Comment:
How about encapsulating the AWS credentials management in AWSCredentialsService? Something like awsCredentialsService.getCredentials().

Reply:
Refactored the credential service. Turns out its credentials must be different than AWS credentials - datajobs.aws.accessKeyId and secretAccessKey used by the control service, because if the K8S cluster is hosted on a EKS instance, some API calls to the instance made by the control-service could fail.

Reply:
My idea is to get rid of AWS credentials/data/properties from the JobImageBuilder and provide the following interface:

```
AwsCredentials/Properties awsCredentials = awsCredentialsService.getCredentials();
```
getCredentials() will be responsible to return the correct AWS credentials/data/properties based on the certain checks which currently are distributed between JobImageBuilder and AwsCredentialsService. Also, the AwsCredentials will contain all related to the AWS data.

```
AwsCredentials.getRegion()
AwsCredentials.getSecret()
AwsCredentials.getKey()
...
etc.
```

In this way, you will encapsulate the all related to AWS credentials logic.